File: packages/globalLibrary/appConstants/src/parse/parseModelsHelper.ts
export class ParseModelsHelper {
  static getUniqueId(map: IParseObject) {
    const id = map.id
    let uniqueId = map.get('uniqueId') || id
    if (!!uniqueId && uniqueId === '') {
      uniqueId = id
    }
    return uniqueId
  }

  static getSyncPostedAt(map: IParseObject) {
    const updatedAt = map.get('updatedAt')
    const syncPostedAt = map.get('syncPostedAt')
    if (syncPostedAt === undefined) {
      return updatedAt
    }
    if (!!syncPostedAt) {
      return syncPostedAt
    }
    return updatedAt
  }
}

File: packages/appFlutter/lib/src/appModels/models/helper/ParseModelsHelper.dart
class ParseModelsHelper {
  static getUniqueId(Map<String, dynamic> json) {
    String uniqueId = json['uniqueId'] as String;
    return uniqueId;
  }

  static getSyncPostedAt(Map<String, dynamic> json) {
    var updatedAt = json['updatedAt'] as String;
    var syncPostedAt = json['syncPostedAt'] as String;
    if (syncPostedAt == null) {
      return updatedAt;
    }
    if (syncPostedAt != null) {
      return syncPostedAt;
    }
    return updatedAt;
  }
}
